### PR TITLE
Added support for MySQL Update Statements

### LINKED ISSUE #2832: Parse MySQL multi-table UPDATE
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

SQLFluff is unable to parse [MySQL multiple table `UPDATE` statements](https://dev.mysql.com/doc/refman/5.7/en/update.html).

### Expected Behaviour

SQLFluff to parse the valid, executable SQL and report on any rule violations.

### Observed Behaviour

> Line 1, Position 1: Found unparsable section: 'UPDATE a, b SET a.name = b.name WHERE a....

### How to reproduce

With `multi-update.sql` as
```sql
UPDATE a, b SET a.name = b.name WHERE a.id = b.id;
```

running SQLFluff results in a parse error:

```sh
$ sqlfluff parse --verbose --ignore-local-config --dialect mysql multi-update.sql 
==== sqlfluff ====
sqlfluff:               0.11.0 python:                 3.8.10
implementation:        cpython dialect:                 mysql
verbosity:                   1 templater:               jinja

=== [ path: multi-update.sql ] ===

[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        update_statement:
[L:  1, P:  1]      |            unparsable:                                       !! Expected: 'UpdateStatementSegment'
[L:  1, P:  1]      |                keyword:                                      'UPDATE'
[L:  1, P:  7]      |                whitespace:                                   ' '
[L:  1, P:  8]      |                raw:                                          'a'
[L:  1, P:  9]      |                comma:                                        ','
[L:  1, P: 10]      |                whitespace:                                   ' '
[L:  1, P: 11]      |                raw:                                          'b'
[L:  1, P: 12]      |                whitespace:                                   ' '
[L:  1, P: 13]      |                raw:                                          'SET'
[L:  1, P: 16]      |                whitespace:                                   ' '
[L:  1, P: 17]      |                raw:                                          'a'
[L:  1, P: 18]      |                raw:                                          '.'
[L:  1, P: 19]      |                raw:                                          'name'
[L:  1, P: 23]      |                whitespace:                                   ' '
[L:  1, P: 24]      |                raw:                                          '='
[L:  1, P: 25]      |                whitespace:                                   ' '
[L:  1, P: 26]      |                raw:                                          'b'
[L:  1, P: 27]      |                raw:                                          '.'
[L:  1, P: 28]      |                raw:                                          'name'
[L:  1, P: 32]      |                whitespace:                                   ' '
[L:  1, P: 33]      |                raw:                                          'WHERE'
[L:  1, P: 38]      |                whitespace:                                   ' '
[L:  1, P: 39]      |                raw:                                          'a'
[L:  1, P: 40]      |                raw:                                          '.'
[L:  1, P: 41]      |                raw:                                          'id'
[L:  1, P: 43]      |                whitespace:                                   ' '
[L:  1, P: 44]      |                raw:                                          '='
[L:  1, P: 45]      |                whitespace:                                   ' '
[L:  1, P: 46]      |                raw:                                          'b'
[L:  1, P: 47]      |                raw:                                          '.'
[L:  1, P: 48]      |                raw:                                          'id'
[L:  1, P: 50]      |    statement_terminator:                                     ';'
[L:  1, P: 51]      |    newline:                                                  '\n'

==== parsing violations ====
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'UPDATE a, b SET
                       | a.name = b.name WHERE a....'
```

### Dialect

MySQL

### Version

sqlfluff, version 0.11.0
Python 3.8.10

### Configuration

All defaults

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by WittierDinosaur**
### Brief summary of the change made
Adds support for MySQL's Update Statements.

Fixes #2832

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: src/sqlfluff/dialects/dialect_mysql.py [hunk 1]
```diff
@@ -1809,3 +1812,23 @@ class OptimizeTableStatementSegment(BaseSegment):
...
+
+class UpdateStatementSegment(BaseSegment):
+    """An `Update` statement.
+
+    As per https://dev.mysql.com/doc/refman/8.0/en/update.html
+    """
+
+    type = "update_statement"
+    match_grammar: Matchable = StartsWith("UPDATE")
+    parse_grammar: Optional[Matchable] = Sequence(
```
Comment (tunetheweb) [inline src/sqlfluff/dialects/dialect_mysql.py:1825]: There's no benefit to the `StartsWith` when it's the first part of `parse_grammar`, and there's no terminatro clause.

```suggestion
    match_grammar: Matchable = Sequence(
```

I agree ANSI, Exasol, and Teradata all have this, but TSQL doesn't. Fancy making them consistent?

**Review by tunetheweb — APPROVED** [review]
Change is good, but personally I'd merge the `match_grammar` and `parse_grammar`.

File: src/sqlfluff/dialects/dialect_mysql.py (hunk 1 above)
Comment (WittierDinosaur) [reply src/sqlfluff/dialects/dialect_mysql.py]: Sorted 👍

**Review by tunetheweb — APPROVED** [review]
Thanks!

File: src/sqlfluff/dialects/dialect_mysql.py [hunk 2]
```diff
@@ -4,6 +4,8 @@
 https://dev.mysql.com/doc/refman/8.0/en/differences-from-ansi.html
 """
 
+from typing import Optional
+
```
Comment (tunetheweb) [inline src/sqlfluff/dialects/dialect_mysql.py:8]:
```suggestion
```

**Pull request merged**